### PR TITLE
Add colorful-mode recipe.

### PR DESCRIPTION
**Pull request by DevelopmentCool2449**
### Brief summary of what the package does

Allow preview color hex and color names, in current buffer at real time,
changing colors to other formats like emacs color names and hex, and adding
user friendly options based on rainbow-mode.el.

### Direct link to the package repository

https://github.com/DevelopmentCool2449/colorful-mode

### Your association with the package

I'm the author and current mantainer.

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

### CODE REVIEW

File: recipes/colorful-mode [hunk 1]
```diff
@@ -0,0 +1,4 @@
+(colorful-mode
+ :fetcher github
+ :repo "DevelopmentCool2449/colorful-mode"
+ :url "https://github.com/DevelopmentCool2449/colorful-mode")
```
Comment (riscy) [inline recipes/colorful-mode:4]:
```suggestion
(colorful-mode
 :fetcher github
 :repo "DevelopmentCool2449/colorful-mode")
```

**Review by riscy — COMMENTED** [review]
Thanks for this, and thank you for specifically calling out rainbow-mode in your README - that was going to be my first question. :)  The `:url` needs to be removed from the recipe, and here's a quick first pass otherwise --

`colorful-mode.el` with byte-compile using Emacs 29.2:
```
colorful-mode.el:541:17: Warning: reference to free variable `global-colorful-modes'
```

`colorful-mode.el` with package-lint 20240330.1458:
```
2 issues found:
1:0: warning: The package summary should not end with a period.
443:5: warning: Closing parens should not be wrapped onto new lines.
```

Package and license:
- Other (via API)
  - Try to use a standard license file format
    See: https://github.com/licensee/licensee
    e.g. https://www.gnu.org/licenses/gpl-3.0.txt

**Comment by DevelopmentCool2449** [comment]
Thank you, i've now update the package and fixed those issues

**Comment by DevelopmentCool2449** [comment]
Hello, i want to know if there is still interest in adding this to Melpa.
Thanks.

**Comment by riscy** [comment]
Hi @DevelopmentCool2449 - I'm here on weekends, so you can typically expect that turnaround time. :)

I notice you've added some new copyright boilerplate, including:

```
;; This file is part of GNU Emacs.
```
which is incorrect (this should be removed) as well as
```
;; Copyright (C) 2024 Free Software Foundation, Inc
```
which is also incorrect I believe.  Unless you've done specific paperwork with the FSF for this package, which I suspect isn't the case, then the copyright is already implicitly your own.  (You can still write it explicitly, but it's not the FSF's)

**Comment by DevelopmentCool2449** [comment]
> Hi @DevelopmentCool2449 - I'm here on weekends, so you can typically expect that turnaround time. :)
> 
> I notice you've added some new copyright boilerplate, including:
> 
> ```
> ;; This file is part of GNU Emacs.
> ```
> 
> which is incorrect (this should be removed) as well as
> 
> ```
> ;; Copyright (C) 2024 Free Software Foundation, Inc
> ```
> 
> which is also incorrect I believe. Unless you've done specific paperwork with the FSF for this package, which I suspect isn't the case, then the copyright is already implicitly your own. (You can still write it explicitly, but it's not the FSF's)

I've done paperwork and i've sent a request for submission to ELPA, i'm waiting the approval from ELPA guys so that is why i had to change the copyright note.

**Comment by riscy** [comment]
> I've done paperwork and i've sent a request for submission to ELPA, i'm waiting the approval from ELPA guys so that is why i had to change the copyright note.

Thank you.  Knowing that you *have* done the paperwork, are you sure you want to host this package on both ELPA and MELPA?  MELPA often avoids indexing packages that are also in GNU ELPA to prevent confusion (e.g. the code would exist both in your GitHub repo and in the Savannah git).

**Comment by DevelopmentCool2449** [comment]
> > I've done paperwork and i've sent a request for submission to ELPA, i'm waiting the approval from ELPA guys so that is why i had to change the copyright note.
> 
> Thank you. Knowing that you _have_ done the paperwork, are you sure you want to host this package on both ELPA and MELPA? MELPA often avoids indexing packages that are also in GNU ELPA to prevent confusion (e.g. the code would exist both in your GitHub repo and in the Savannah git).

Sure, host it in MELPA would allow to people that want to get newest updates from my package, and ELPA only for stable releases.

**Comment by riscy** [comment]
Okay - the last thing you ought to check (remove) is this line:

`;; This file is part of GNU Emacs.`

This isn't true since - even though you're planning on hosting the package on ELPA - the package isn't distributed with Emacs core (although maybe some day it will be).  This could create some confusion for users if you leave it in.

**Comment by DevelopmentCool2449** [comment]
I thought about it a lot and since this package is now in ELPA, I don't think it would be convenient to add it to MELPA, anyway thanks for the advices.

**Pull request closed without merging**